onnxruntime/core/session/custom_ops.cc

```diff
@@ -119,8 +119,19 @@ common::Status CreateCustomRegistry(const std::vector<OrtCustomOpDomain*>& op_do
       KernelDefBuilder def_builder;
       def_builder.SetName(op->GetName(op))
           .SetDomain(domain->domain_)
-          .SinceVersion(1)
-          .Provider(onnxruntime::kCpuExecutionProvider);
+          .SinceVersion(1);
+      if (&op->GetExecutionProviderType) {
+        auto provider = op->GetExecutionProviderType(op);
+        if (provider == ORT_EXECUTION_PROVIDER_CPU)
```

Comment:
it will be good to have a function to define the mapping between execution provider string and enum.

Reply:
I have put this into `GetProviderString` function in custom_ops.cc. Please check the latest commit.